go-controller/pkg/ovn/ovn.go

```diff
@@ -716,23 +599,34 @@ func (oc *Controller) WatchPods() {
 					podNs, podName)
 				return
 			}
-			if !oc.ensurePod(oldPod, pod, oc.checkAndSkipRetryPod(pod.UID)) {
+			if err := oc.ensurePod(oldPod, pod, oc.checkAndSkipRetryPod(pod)); err != nil {
```

Comment:
we also need to check in the cache if there is a delete entry in the retry here and delete it it before we add the new pod. We could get into a scenario like:
1. pod foo/bar exists with 10.0.0.5 on node A
2. delete for pod foo/bar fails, add to retry delete
3. pod foo/bar is recreated on node B, fails to be added
4. pod foo/bar gets updated, now we add it here and it gets 10.0.1.6, delete the entry in retry cache, now we never clean up the IPs/routes from 1

Reply:
ack, waiting for https://github.com/ovn-org/ovn-kubernetes/pull/2826/files#r809622862 to merge and I can do the same-ish thing here and in add above.